pkg/sql/opt/optbuilder/select.go

```diff
@@ -268,6 +268,13 @@ func (b *Builder) buildDataSource(
 
 		lockCtx.filter(source.As.Alias)
 		if lockCtx.locking.isSet() {
+			// If this table was on the null-extended side of an outer join, we are not
+			// allowed to lock it.
+			if lockCtx.isNullExtended {
+				panic(pgerror.Newf(
+					pgcode.FeatureNotSupported, "%s cannot be applied to the nullable side of an outer join",
```

Comment:
should this link to a github issue using the `unimplemented` package, or no need?

Reply:
I don't think so, since this behavior matches Postgres. It's not something we're planning to change.

Reply:
makes sense, thanks for clarifying!